src/AutoRest.CSharp/Mgmt/Decorator/TypeExtensions.cs

```diff
@@ -72,29 +62,20 @@ public static CSharpType WrapOperation(this CSharpType type, bool isAsync)
             return isAsync ? new CSharpType(typeof(Task<>), response) : response;
         }
 
-        public static CSharpType WrapResponse(this Type type, bool isAsync)
+        public static bool TryCast<T>(this CSharpType type, [MaybeNullWhen(false)] out T provider) where T : TypeProvider
```

Comment:
Can we move this to the CSharpType class as a member? I believe this is very useful